Add absolute fee rate convenient functions to `FeeRate`

tcharding opened the pull request:
- Patch 1 is docs cleanup
- Patch 2 adds two functions to `FeeRate`

From the commit log of patch 2:
    
    Calculating the absolute fee from a fee rate can currently be achieved
    by creating a `Weight` object and using
    `FeeRate::checked_mul_by_weight`. This is kind of hard to discover, we
    can add two public convenience functions that make discovery of the
    functionality easier.
    
    Add two functions for calculating the absolute fee by multiplying by
    weight units (`Weight`) and virtual bytes (by first converting to weight
    units).


This seems like an obvious thing so I'm inclined to think that @Kixunil left it out for a reason. (Mentioning you here Kix so even if this merges you'll see it in notifications later on.)

apoelstra reviewed (dismissed):
ACK 89c81193a71eaf6b55b0e09383c7188ee6b973ab

yancyribbens commented on bitcoin/src/blockdata/fee_rate.rs:
```diff
@@ -87,6 +87,23 @@ impl FeeRate {
+    /// `None` if overflow occurred.
+    ///
+    /// This is equivalent to `Self::checked_mul_by_weight()`.
+    pub fn absolute_fee_wu(self, weight: Weight) -> Option<Amount> {
```
Also, why `absolute fee` instead of `fee_wu`?  if the name `checked_mul_by_weight` is confusing, maybe just rename that function instead of adding this wrapper function?

danielabrozzoni replied:
Personally I like the `absolute_fee_wu` name, I've always found bdk's `fee_wu` to be quite confusing. I don't have any strong opinion on renaming vs adding a wrapper.

yancyribbens replied:
What does the work `absolute` mean in the context of fee?  Is there such a thing as a non-absolute fee?

tcharding replied:
I dug around math and english definitions of "absolute" and, contrary to my gut feeling, it does not add any additional meaning although it was also [mis]used in [this Bitocin Core PR](https://github.com/bitcoin/bitcoin/pull/16507/commits/8e59af55aaf1b196575084bce2448af02d97d745).

 I've removed the `absolute_` prefix.

danielabrozzoni reviewed (approved):
ACK 89c81193a71eaf6b55b0e09383c7188ee6b973ab

tcharding commented:
Changes in force push:
- Added an additional trivial rustdoc fix patch (as patch 2)
- Rename functions by dropping the "absolute_" prefix
- Added a unit test to verify the two convenience functions agree

sanket1729 reviewed (approved):
utACK 9eff2f2f5eb7e00bc646477bba3ace0467f244f4

yancyribbens commented on bitcoin/src/blockdata/fee_rate.rs:
```diff
@@ -199,4 +225,20 @@ mod tests {
     }
+
+    #[test]
+    fn fee_convenience_functions_agree() {
```
this test could be boilled down to like two lines:

```
fn fee_convenience_functions_agree() {
    let rate = FeeRate::from_sat_per_vb(1).expect("1 sat/byte is valid");
    assert_eq!(rate.fee_vb(1), rate.fee_wu(Weight::from_wu(4)));                           
}
```

after all, no need to test all of the transaction serialization stuff again here.

apoelstra reviewed (approved):
ACK 9eff2f2f5eb7e00bc646477bba3ace0467f244f4

apoelstra commented:
Removed the @ mention from the PR description because it's likely to cause more noise than intended.

tcharding commented:
Oh I thought it only spammed if in the commit not in the PR?

apoelstra commented:
@tcharding the merge script copies the PR message into the commit.

tcharding commented:
Oh of course. Thanks.
